### ISSUE TITLE
Unable to install svc-cat/catalog using Helm

### ISSUE THREAD

**Issue by oke-py** [kubernetes-sigs/service-catalog#2787]
# Bug Report 

**What happened**:

Following [this document](https://kubernetes.io/docs/tasks/service-catalog/install-service-catalog-using-helm/#install-service-catalog-in-your-kubernetes-cluster), I executed this command but it failed.

```
$ helm install catalog svc-cat/catalog --namespace catalog
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1"
```

**What you expected to happen**:

svc-cat/catalog is successfully installed.

**How to reproduce it (as minimally and precisely as possible)**:

See "What happened".

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T23:41:24Z", GoVersion:"go1.14", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:07:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```
- service-catalog version:
- Cloud provider or hardware configuration:
- Do you have api aggregation enabled? 
  - Do you see the configmap in kube-system? 
  - Does it have all the necessary fields?
    - `kubectl get cm -n kube-system extension-apiserver-authentication -o yaml` and look for `requestheader-XXX` fields
- Install tools:
  - Did you use helm? What were the helm arguments? Did you `--set` any extra values?
- Are you trying to use ALPHA features? Did you enable them?

**Comment by mszostok** [comment]
Hi @oke-py 

the problem is that the last release do not support the K8s in version > 1.16 
Here is the pr which was fixing that issue: https://github.com/kubernetes-sigs/service-catalog/pull/2753

I will try to do the release this week to solve such problems. 

As a temporary workaround, you can just install Service Catalog from master:
```
# clone repo
git clone https://github.com/kubernetes-sigs/service-catalog.git

cd service-catalog

# install with sc image build from the latest commit on master
helm install charts/catalog \
    --set image=quay.io/kubernetes-service-catalog/service-catalog:v0.3.0-beta.2-16-g52979db \
    --name=service-catalog \
    --wait
```

if you want to can also use `--set image=quay.io/kubernetes-service-catalog/service-catalog:canary` to use latest master.

**Comment by oke-py** [comment]
I got it. Thanks!

**Comment by cmoulliard** [comment]
When a new release is scheduled please as this problem is blocking ?

**Comment by rajshkr** [comment]
> Hi @oke-py
> 
> the problem is that the last release does not support the K8s in version >= 1.16
> Here is the pr which was fixing that issue: #2753
> 
> but this is not only available on master. I will try to do the release this week to solve such problems.
> 
> As a temporary workaround, you can just install Service Catalog from master:
> 
> ```
> # clone repo
> git clone https://github.com/kubernetes-sigs/service-catalog.git
> 
> cd service-catalog
> 
> # install with sc image build from the latest commit on master
> helm install charts/catalog \
>     --set image=quay.io/kubernetes-service-catalog/service-catalog:v0.3.0-beta.2-16-g52979db \
>     --name=service-catalog \
>     --wait
> ```
> 
> if you want to can also use `--set image=quay.io/kubernetes-service-catalog/service-catalog:canary` to use latest master.

I tried to install the release using the above suggestion and helm install failed. Any idea what's went wrong?

**Comment by mszostok** [comment]
Hi @rajshkr 
could you share the error message? 

maybe it was because of today's quay.io outage 😞 

<img width="1386" alt="obraz" src="https://user-images.githubusercontent.com/17568639/82374183-f8145580-9a1e-11ea-83b0-91cb8781f481.png">


Check https://status.quay.io/ for more info

**Comment by rajshkr** [comment]
> Hi @rajshkr
> could you share the error message?
> 
> maybe it was because of today's quay.io outage 😞
> 
> <img alt="obraz" width="1386" src="https://user-images.githubusercontent.com/17568639/82374183-f8145580-9a1e-11ea-83b0-91cb8781f481.png">
> 
> Check https://status.quay.io/ for more info

Yes.  I tried now and it is working now. Have deployed service catlog and minibroker as well.  I am now seeing a different issue when creating a service instance

**kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/service-catalog/master/contrib/examples/walkthrough/mini-instance.yaml**
**svcat describe instance -n test-ns mini-instance**
  Name:        mini-instance
  Namespace:   test-ns
  Status:      OrphanMitigation - Provision call failed: service instance "7ed9d8f1-e024-4747-be77-4bfeb825ff9d" failed to provision @ 2020-05-20 04:04:37 +0000 UTC
  Class:       mariadb
  Plan:        10-1-26

Parameters:
  param-1: value-1
  param-2: value-2

Bindings:
No bindings defined

Another observation 

**svcat describe plan 10-1-26 --scope cluster**
Error: unable to list cluster-scoped plans (field label not supported: spec.externalName)

**Comment by mszostok** [comment]
Hi @rajshkr

which `svcat` version are you using? It looks like an old one.

Btw. Today we released the 0.3.0 Service Catalog version which supports the k8s 1.18. 

I also tested that flow with the k8s 1.18 using kind (`kind create cluster --image="kindest/node:v1.18.2"`) and I spot one issue.

Currently when you are executing:
```
kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/service-catalog/master/contrib/examples/walkthrough/mini-instance.yaml
``` 

you are creating mariadb with old version which does not support k8s 1.18.  Here is pr to fix that bug: https://github.com/kubernetes-sigs/service-catalog/pull/2812

Currently, you can just execute:

```
kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/service-catalog/92a74067df77e5112107a39dacd3a839d342a1db/contrib/examples/walkthrough/mini-instance.yaml
```

**Comment by mszostok** [comment]
Hi @cmoulliard

ICYMI: The new release is available, here you can find more info :) https://kubernetes.slack.com/archives/C232SF3TK/p1590424684027100

**Comment by oke-py** [comment]
It works!
```
$ kubectl version          
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:16:24Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-30T20:19:45Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl create ns catalog
namespace/catalog created

$ helm install catalog svc-cat/catalog --namespace catalog
NAME: catalog
LAST DEPLOYED: Tue May 26 14:27:03 2020
NAMESPACE: catalog
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

**Comment by cmoulliard** [comment]
I did a test successfully

**Comment by rajshkr** [comment]
> Hi @rajshkr
> 
> which `svcat` version are you using? It looks like an old one.
> 
> Btw. Today we released the 0.3.0 Service Catalog version which supports the k8s 1.18.
> 
> I also tested that flow with the k8s 1.18 using kind (`kind create cluster --image="kindest/node:v1.18.2"`) and I spot one issue.
> 
> Currently when you are executing:
> 
> ```
> kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/service-catalog/master/contrib/examples/walkthrough/mini-instance.yaml
> ```
> 
> you are creating mariadb with old version which does not support k8s 1.18. Here is pr to fix that bug: #2812
> 
> As a workaround use that command (after merging my pr this workaroud will be not needed):
> 
> ```
> kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/service-catalog/92a74067df77e5112107a39dacd3a839d342a1db/contrib/examples/walkthrough/mini-instance.yaml
> ```
> 
> Please ping me on k8s slack (nick: mszostok) if you will have any other prroblems

I upgraded to 0.3.0 of svc and used the supported version of mariaDB. It worked as a charm!!
